sdk/identity/azure-identity/azure/identity/_credentials/azure_cli.py

```diff
@@ -136,10 +143,17 @@ def get_token_info(self, *scopes: str, options: Optional[TokenRequestOptions] =
         :rtype: ~azure.core.credentials.AccessTokenInfo
         :return: An AccessTokenInfo instance containing information about the token.
 
-        :raises ~azure.identity.CredentialUnavailableError: the credential was unable to invoke the Azure CLI.
+        :raises ~azure.identity.CredentialUnavailableError: the credential was either unable to invoke the Azure CLI
+          or a claims challenge was provided.
         :raises ~azure.core.exceptions.ClientAuthenticationError: the credential invoked the Azure CLI but didn't
           receive an access token.
         """
+        claims_value = options.get("claims") if options else None
+        if claims_value and claims_value.strip():
+            login_cmd = f"az login --claims-challenge {claims_value}"
+            if scopes:
+                login_cmd += f" --scope {scopes[0]}"
+            raise CredentialUnavailableError(f"Failed to get token. Run {login_cmd}")
```

Comment:
Let's add this logic to `_get_token_base`

Reply:
Also, add "--tenant <tenant_id>" to the login command if a tenant_id was passed into the options.